src/Command/PopulateIndex.php

```diff
@@ -67,64 +65,67 @@ private function getIndex(): ?IndexInterface
         return null;
     }
 
-    private function populateIndex(IndexInterface $indexConfig, ElasticaIndex $esIndex): void
+    private function populateIndex(IndexInterface $indexConfig, ElasticaIndex $esIndex): int
     {
-        ProgressBar::setFormatDefinition('custom', "%percent%%\t%remaining%\t%memory%\n%message%");
-
-        $progressBar = new ProgressBar($this->output, 1);
-        $progressBar->setMessage('');
-        $progressBar->setFormat('custom');
-
         try {
             foreach ($indexConfig->getAllowedDocuments() as $document) {
-                $progressBar->setProgress(0);
-                $progressBar->setMessage($document);
-
                 $documentInstance = $this->documentRepository->get($document);
 
                 $this->documentHelper->setTenantIfNeeded($documentInstance, $indexConfig);
 
                 $listingCount = $documentInstance->getListingInstance($indexConfig)->count();
-                $progressBar->setMaxSteps($listingCount > 0 ? $listingCount : 1);
-                $esDocuments = [];
                 $numberOfBatches = ceil($listingCount / $indexConfig->getBatchSize());
 
-                for ($batchNumber = 0; $batchNumber < $numberOfBatches; $batchNumber++) {
-                    $listing = $documentInstance->getListingInstance($indexConfig);
-                    $listing->setOffset($batchNumber * $indexConfig->getBatchSize());
-                    $listing->setLimit($indexConfig->getBatchSize());
-
-                    foreach ($listing->getData() ?? [] as $dataObject) {
-                        try {
-                            $progressBar->advance();
-
-                            if (!$documentInstance->shouldIndex($dataObject)) {
-                                continue;
-                            }
-
-                            $esDocuments[] = $this->documentHelper->elementToDocument($documentInstance, $dataObject);
-                        } catch (\Throwable $throwable) {
-                            $this->displayDocumentError($indexConfig, $document, $dataObject, $throwable);
+                $this->output->getFormatter()->setDecorated(true);
+                $this->output->writeln('');
+
+                if (!$indexConfig->shouldIndexInSubprocesses()) {
+                    $this->output->writeln(
+                        '<info>For large indexes please consider to implement `shouldIndexInSubprocesses` to prevent memory exhaustion.',
+                    );
```

Comment:
I like the idea, but maybe we can find a simple metric when to display this message? E.g. if in dev mode and the number of documents for a given index exceeds 25k?

Reply:
I thought about that and still open to that. Its hard to define a metric since it also depends on the documents to index. I would show it always in dev mode independent from the index size. What do you think?

Reply:
As discussed I set this to 10000k